### ISSUE TITLE
Specification: Integration tests

### ISSUE THREAD

**Issue by jnehring** [freme-project/technical-discussion#29]
We need automated integration tests: These tests test each API endpoint. Each test performs API calls via HTTP and evaluates the output. Test framework will be JUnit, all tests will be implemented in the Broker.

# Joint test environment across all tests

We want to setup test environment (start broker) only once, then execute the whole test suite and tear down test environment afterwards. Right now the unit tests are executed using maven surefire plugin. I believe for integration tests the [Maven failsafe plugin](http://maven.apache.org/surefire/maven-failsafe-plugin/) is better because it supports setting up a joint test environment across several test classes. An extensive discussion about this topic is [here](http://stackoverflow.com/questions/82949/before-and-after-suite-execution-hook-in-junit-4-x). An alternate solution discussed there is to use maven surefire plugin and register a "RunListener" to manage joint test environment but this sounds like a dirty hack - maven failsafe plugin sounds way better.

# Standalone test tool

A nice to have feature would be to export a runnable jar from the test cases. We can then use this jar to verify any currently running FREME API which is especially useful to verify that the live installation of FREME works correct. That seams to be quiet easy as described [here](http://stackoverflow.com/questions/4648341/how-to-export-junit-test-suite-as-executable-jar). The base url for the api under test can be passed as a command line parameter. This feature can provide a basis for constant monitoriong of FREME API.

# Timeline and feature set for FREME 0.2

In FREME 0.2 we will create the framework for integration testing as described here. Then in FREME 0.3 we will program integration tests for all api endpoints. The reason for not doing the complete work package in FREME 0.2 is
* We already have a long feature list for FREME 0.2
* We are going to hire students that start 1.7. / 15.7.. They will program the API tests. They start too late to do this work.
